pkg/install/upgrade.go

```diff
@@ -14,12 +14,32 @@ import (
 )
 
 func (i *Installer) upgradeCluster(ctx context.Context) error {
+	vsn, err := version.ParseVersion(version.OpenShiftVersion)
+	if err != nil {
+		return err
+	}
+
 	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
 		cv, err := i.configcli.ConfigV1().ClusterVersions().Get("version", metav1.GetOptions{})
 		if err != nil {
 			return err
 		}
 
+		if cv.Spec.Channel != "" {
+			i.log.Printf("not upgrading: cvo channel is %s", cv.Spec.Channel)
+			return nil
+		}
+
+		desired, err := version.ParseVersion(cv.Status.Desired.Version)
+		if err != nil {
+			return err
+		}
+
+		if !desired.Lt(vsn) {
```

Comment:
Just a question here about the ability to downgrade.  Is there a scenario where we would want to downgrade?

Reply:
It can be done manually if needed.